Sources/MapboxCoreNavigation/Navigator.swift

```diff
@@ -1,7 +1,110 @@
 import Foundation
 import MapboxNavigationNative
+import MapboxDirections
 
 extension Navigator {
+    
+    /**
+     Tiles version string. If not specified explicitly - will be automatically resolved
+     to the latest version.
+     
+     This property can only be modified before creating `Navigator` shared instance, all
+     further changes to this property will have no effect.
+     */
+    static var tilesVersion: String = ""
+    
+    /**
+     A local path to the tiles storage location. If not specified - will be automatically defaulted
+     to the cache subdirectory.
+     
+     This property can only be modified before creating `Navigator` shared instance, all
+     further changes to this property will have no effect.
+     */
+    static var tilesURL: URL? = nil
+    
+    /**
+     Shared instance on `Navigator`.
+     */
+    static let shared: Navigator = {
+        return navigatorWithHistoryRecorder.0
+    }()
```

Comment:
I should’ve been clearer in https://github.com/mapbox/mapbox-navigation-ios/pull/2823#discussion_r582147652: the suggestion was to use composition/encapsulation rather than an extension. `MapboxCoreNavigation.Navigator` would be a singleton containing a `MapboxNavigationNative.Navigator`. This way the raw methods defined by MapboxNavigationNative won’t be exposed to the developer. It’ll also be possible to encapsulate other resources at the same time, such as HistoryRecorderHandle – obviating both the `NavigationResources` struct from 5ac8c9a5ddebd4f90f49ab391c54ac9ff57707df and the `(Navigator, HistoryRecorderHandle)` tuple defined in this PR.

Reply:
Got it. Please check latest update, `CoreNavigationNavigator.swift` now encapsulates both `HistoryRecorderHandle` and `MapboxNavigationNative.Navigator`. Access point is possible via `Navigator.shared` static computed property.